lib/httparty/request.rb

```diff
@@ -92,6 +92,11 @@ def uri
 
         path.path = "/#{path.path}" if last_uri_host[-1] != "/"
         path.path = last_uri_host + path.path
+
+        # use absolute path [#582]
+        if path.path.include?('/../')
```

Comment:
Should we account for paths like `./foo/bar` as well?

Reply:
It's a slightly different case but happy to tackle it if you think it's worth it?  

Given a request to https://test.com/v1
 
a redirect to a relative path of "../foo/bar" should result in a new request to https://test.com/foo/bar - the foo directory is a sibling of /v1

a redirect to a  relative path of "./foo/bar" should result in a new request to https://test.com/v1/foo/bar - the foo directory is a child of v1